packages/block-library/src/gallery/style.scss

```
@@ -1,71 +1,67 @@
-.wp-block-gallery,
-.blocks-gallery-grid {
-	display: flex;
-	flex-wrap: wrap;
-	list-style-type: none;
-	padding: 0;
-	// Some themes give all <ul> default margin instead of padding.
-	margin: 0;
-
-	.blocks-gallery-image,
-	.blocks-gallery-item {
-		// Add space between thumbnails, and unset right most thumbnails later.
-		margin: 0 1em 1em 0;
+.wp-block-gallery {
+	ul.blocks-gallery-grid {
 		display: flex;
-		flex-grow: 1;
-		flex-direction: column;
-		justify-content: center;
-		position: relative;
-
-		// On mobile and responsive viewports, we allow only 1 or 2 columns at the most.
-		width: calc(50% - 1em);
-
-		&:nth-of-type(even) {
-			margin-right: 0;
-		}
+		flex-wrap: wrap;
+		list-style-type: none;
+		padding: 0;
+		// Some themes give all <ul> default margin instead of padding.
+		margin: 0;
+
+		li.wp-block-image {
+			// Add space between thumbnails, and unset right most thumbnails later.
+			margin: 0 $grid-unit-20 $grid-unit-20 0;
+			display: flex;
+			flex-grow: 1;
+			flex-direction: column;
+			justify-content: center;
+			position: relative;
 
```

The above margin rule overrides the removal of the right margin on the last gallery image. Perhaps we can just move it to here so it will take precedence.
```suggestion
			&:last-child {
			    margin-right: 0;
			}
```

---

Thanks, that fixed the margin issue!